packages/react-debug-tools/src/ReactDebugHooks.js

```diff
@@ -402,24 +412,53 @@ function buildTree(rootStack, readHookLog): HooksTree {
       subHooks: [],
     });
   }
+
+  // Associate custom hook values (useInpect() hook entries) with the correct hooks
+  rootChildren.forEach(hooksNode => rollupDebugValues(hooksNode));
+
   return rootChildren;
 }
 
+function rollupDebugValues(hooksNode: HooksNode): void {
+  let useInpectHooksNodes: Array<HooksNode> = [];
+  hooksNode.subHooks = hooksNode.subHooks.filter(subHooksNode => {
+    if (subHooksNode.name === 'DebugValue') {
```

Comment:
Maybe we should tag these nodes with something more unique to the built-in one so that it doesn't happen with custom hooks that happen to share the same name. E.g. You could also check that this is the terminal node. children.length === 0

Reply:
Good point. I meant to do this but forgot. 😁

Reply:
This seems trickier than I thought because of how we're parsing the stack for function names and then comparing them to the primitive labels.

For example, I can name the local function something more unique (e.g.  `useDebugValue__REACT_INTERNAL`) but if it's assigned to the dispatcher as `useDebugValue`– then function name that's being parsed and passed to `isReactWrapper` is `useDebugValue`.

Maybe I'm overlooking something obvious. I'll come back to this.